### PR TITLE
Pub-1030 - Connect Court A-Z page with Reference data endpoint 

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link ###

https://tools.hmcts.net/jira/browse/PUB-1030

### Change description ###

These changes will get the court data from data management api endpoint. All filters are being populated from reference data now.

NOTE: We need to remove some e2e tests now because we will keep getting reference data from the different source which cause total no of courts to be changed. Same for hearing and live case status, there is a possibility that we do not have hearing for a specific court which is defined in e2e tests.

Image 11-02-2022 at 15 30

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

**Review by kurt-uk — CHANGES_REQUESTED** [review]

**Review by marcogagliardi — APPROVED** [review]
👌🏽

File: src/main/controllers/AlphabeticalSearchController.ts [hunk 1]
```diff
@@ -37,17 +43,13 @@ export default class AlphabeticalSearchController {
...
     filterValues = Array.prototype.concat.apply([], values);
 
-    keys.forEach(key => {
-      if (key === 'Region') {
-        keys.splice(keys.indexOf(key), 1, 'Location');
-      }
-    });
-
-    const alphabetisedList = await courtService.generateFilteredAlphabetisedCourtList(keys, filterValues);
+    const alphabetisedList = await courtService.generateFilteredAlphabetisedCourtList(filters['Region'], filters['Jurisdiction']);
```
[the remark this answers is not in the document]
Comment (junaidiqbalmoj) [reply src/main/controllers/AlphabeticalSearchController.ts:52]: No, this is how it was build before and I did not change that logic. For filters, we are calling the API again, and it returns us the filtered data. We are not filtering data inside frontend.

File: src/main/controllers/CourtNameSearchController.ts [hunk 2]
```diff
@@ -42,7 +52,7 @@ export default class CourtNameSearchController {
     });
     filterValues = Array.prototype.concat.apply([], values);
 
-    const alphabetisedList = await courtService.generateFilteredAlphabetisedCourtList(keys, filterValues);
+    const alphabetisedList = await courtService.generateFilteredAlphabetisedCourtList(filters['Region'], filters['Jurisdiction']);
```
[the remark this answers is not in the document]
Comment (junaidiqbalmoj) [reply src/main/controllers/CourtNameSearchController.ts:55]: No, this is how it was build before and I did not change that logic. For filters, we are calling the API again, and it returns us the filtered data. We are not filtering data inside frontend.

**Review by kurt-uk — APPROVED** [review]

**Review by marcogagliardi — DISMISSED** [review]
👌🏽

**Review by kurt-uk — APPROVED** [review]

**Review by commitBlob — APPROVED** [review]

**Pull request merged**